src/index.tsx

```diff
@@ -10,109 +8,205 @@ import {
-	commands = app.commands;
-	shell = app.shell;
-
-	const factory = new NCViewerFactory({
-		name: FACTORY_NAME,
-		fileTypes: [FILETYPE],
-		defaultFor: [FILETYPE],
-		readOnly: true
-	});
-
-	let ft: DocumentRegistry.IFileType = {
-		name: FILETYPE,
-		extensions: ['.nc'],
-		mimeTypes: ['application/netcdf'],
-		contentType: 'file',
-		fileFormat: 'base64'
-	}
-
-	app.docRegistry.addFileType(ft);
-	app.docRegistry.addWidgetFactory(factory);
-
-	factory.widgetCreated.connect((sender, widget) => {
-		console.log('NCViewerWidget created from factory');
-	});
-
-	// Creates the left side bar widget when the app has started
-	app.started.then(()=>{
-		// Create the left side bar
-		sidebar = new LeftSideBarWidget(commands, null);
-		sidebar.id = 'vcs-left-side-bar';
-		sidebar.title.label = 'vcs';
-		sidebar.title.closable = true;
-
-		// Attach it to the left side of main area
-		shell.addToLeftArea(sidebar);
-	});
-
-	// Whenever a panel is changed in the shell, this will trigger
-	app.shell.activeChanged.connect((sender,data)=>{
-		if(!data.oldValue && data.newValue && data.newValue.hasClass("jp-NotebookPanel") && !sidebar.notebook){
-			sidebar.updateNotebook(data.newValue);
-		}
-		if(data.oldValue && data.newValue && data.newValue.hasClass("jp-NotebookPanel")){
-			sidebar.updateNotebook(data.newValue);
-		}
-	});
-};
+function activate(app: JupyterLab, tracker: NotebookTracker) {
+  commands = app.commands;
+  shell = app.shell;
+  nb_tracker = tracker;
+
+  const factory = new NCViewerFactory({
+    name: FACTORY_NAME,
+    fileTypes: [FILETYPE],
+    defaultFor: [FILETYPE],
+    readOnly: true
+  });
+
+  let ft: DocumentRegistry.IFileType = {
+    name: FILETYPE,
+    extensions: [".nc"],
+    mimeTypes: ["application/netcdf"],
+    contentType: "file",
+    fileFormat: "base64"
+  };
+
+  app.docRegistry.addFileType(ft);
+  app.docRegistry.addWidgetFactory(factory);
+
+  /*factory.widgetCreated.connect((sender, widget) => {
+    console.log("NCViewerWidget created from factory");
+  });*/
+
+  // Creates the left side bar widget once the app has fully started
+  app.started.then(() => {
+    // Create the left side bar
+    sidebar = new LeftSideBarWidget(commands, nb_tracker);
+    sidebar.id = "vcdat-left-side-bar";
+    sidebar.title.iconClass = "jp-vcdat-icon jp-SideBar-tabIcon";
+    sidebar.title.closable = true;
+
+    // Attach it to the left side of main area
+    shell.addToLeftArea(sidebar);
+
+    // Activate the widget
+    shell.activateById(sidebar.id);
+  });
+
+  // Sets the current notebook once the application shell has been restored
+  // and all the widgets have been added to the notebooktracker
+  app.shell.restored.then(() => {
+    if (nb_tracker.currentWidget instanceof NotebookPanel) {
+      console.log("Currently open notebook selected.");
+      nb_panel_current = nb_tracker.currentWidget;
+      // Check if the notebook had loaded a file already
+      try {
+        let fp: any = nb_utils.getMetaData(
```

Comment:
unless its super obvious what it is, dont use one or two letter variables names. What is "fp"?

Reply:
Sorry, that was an experimental variable that I forgot to rename. Fixed!